File: 01_Language/01_Functions/js/array_push.js
function array_push(arr) {
    if (Array.isArray(arr)) {
        let nextIdx = arr.length
        for (let i = 1; i < arguments.length; i++) {
            arr[nextIdx] = arguments[i]
            nextIdx++
        }
        return arr.length
    }

    let maxIdx = 0
    let length = 0
    for (let i in arr) {
        if (Number.isInteger(i * 1)) {
            let curIdx = i * 1
            if (curIdx > maxIdx) {
                maxIdx = curIdx
            }
        }
        length++
    }
    for (let i = 1; i < arguments.length; i++) {
        maxIdx++
        arr[maxIdx] = arguments[i]
        length++
    }
    return length
}

let stack = ["orange", "banana"]
console.log(array_push(stack, "apple", "raspberry"), stack)

let stack1 = {"k1": "orange", 2: "banana"}
console.log(array_push(stack1, "apple", "raspberry"), stack1)

File: 01_Language/01_Functions/python/array_push.py
# coding: utf-8

def array_push(arr, *args):
    if isinstance(arr, dict):
        max_idx = -1
        for k in arr.keys():
            if isinstance(k, int) and k > max_idx:
                max_idx = k
        for i in args:
            max_idx += 1
            arr[max_idx] = i
    else:
        arr.extend(args)
    return len(arr)


if __name__ == '__main__':
    stack = ["orange", "banana"]
    print(array_push(stack, "apple", "raspberry"))
    print(stack)

    stack = {"k1": "orange", 2: "banana"}
    print(array_push(stack, "apple", "raspberry"))
    print(stack)
